mysql-test/suite/innodb/t/temp_table_encrypt.test

```diff
@@ -0,0 +1,113 @@
+# test temporary tables encryption
+
+--source include/have_innodb.inc
+
+--let $MYSQL_DATA_DIR= `select @@datadir`
+
+CREATE TEMPORARY TABLE t01 (a TEXT) ENGINE=InnoDB;
+INSERT INTO t01 VALUES ('Lorem ipsum dolor sit amet, consectetur adipiscing elit');
+
+# wait until pages flushed
+--let $wait_condition= SELECT COUNT(*) = 0 FROM performance_schema.global_status WHERE LOWER(variable_name) = 'innodb_buffer_pool_pages_dirty' AND variable_value > 0
+--source include/wait_condition.inc
+
+# make sure ibtmp1 is not encrypted
+
+--let $grep_pattern= Lorem ipsum dolor sit amet
+--let $grep_file= $MYSQL_DATA_DIR/ibtmp1
+--let $grep_output= boolean
+--source include/grep_pattern.inc
+
+SET GLOBAL innodb_temp_tablespace_encrypt = ON;
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TEMPORARY TABLE t02 (a INT) ENGINE=InnoDB ENCRYPTION='Y';
+
+CREATE TEMPORARY TABLE t03 (a TEXT) ENGINE=InnoDB ENCRYPTION='N';
+INSERT INTO t03 VALUES ('Curabitur laoreet, velit non interdum venenatis');
+
+# wait until pages flushed
+--let $wait_condition= SELECT COUNT(*) = 0 FROM performance_schema.global_status WHERE LOWER(variable_name) = 'innodb_buffer_pool_pages_dirty' AND variable_value > 0
+--source include/wait_condition.inc
+
+# make sure ibtmp1 is encrypted now
+
+--let $grep_pattern= Curabitur laoreet, velit non interdum venenatis
+--let $grep_file= $MYSQL_DATA_DIR/ibtmp1
+--let $grep_output= boolean
+--source include/grep_pattern.inc
+
+# this table created in separate file per table tablespace
+CREATE TEMPORARY TABLE t04 (a INT) ENGINE=InnoDB ROW_FORMAT=COMPRESSED;
+INSERT INTO t04 VALUES (1), (2), (3);
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TEMPORARY TABLE t05 (a INT) ENGINE=InnoDB ROW_FORMAT=COMPRESSED ENCRYPTION='y';
+
+CREATE TEMPORARY TABLE t06 (a INT) ENGINE=InnoDB ROW_FORMAT=COMPRESSED ENCRYPTION='n';
```

Comment:
And here?

Reply:
Basically I wanted to check that ENCRYPTION='y' is still not allowed for temp tables.